First Task/src/main/java/task/Main.java

```diff
@@ -0,0 +1,18 @@
+package task;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
+public class Main {
+
+    public static void main(String[] args) {
+        ArrayList<Integer> list = new ArrayList<>(Arrays.asList(1, 4, 5, 9));
+        try {
+            System.out.println("Sum is " + Functional.sum(list));
+            System.out.println("Max is " + Functional.max(list));
+        } catch (NoSuchElementException error) {
```

Comment:
You don't need to catch this exception. Let me explain. We know exactly when the exception will be thrown. It will be thrown if the passed list will be empty. And we have a legit way to check if it contains any elements just invoking `isEmpty()` on the list. On the other hand, `try` is pretty costly, it will take more resources to perform the same logic. So this kind of exceptions is easier to handle logically rather than making an attempt in a `try` block and `catch` if it fails.
```java
if (!list.isEmpty()) {
    System.out.println("Sum is " + Functional.sum(list));
    System.out.println("Max is " + Functional.max(list));
} else {
    some other logic
}
```

Well, since you create an instance of a list just in the place you don't even need any checks and error handling, and you can write just:

```java
ArrayList<Integer> list = new ArrayList<>(Arrays.asList(1, 4, 5, 9));
System.out.println("Sum is " + Functional.sum(list));
System.out.println("Max is " + Functional.max(list));
```